rest-client/src/main/java/io/apicurio/registry/client/request/RequestExecutor.java

```diff
@@ -18,15 +18,26 @@
 
 import retrofit2.Call;
 
+import java.util.Map;
+
+
 /**
  * @author Carles Arnal <carles.arnal@redhat.com>
  */
 public class RequestExecutor {
 
+    private final ThreadLocal<Map<String, String>> requestHeaders;
+
+    public RequestExecutor(ThreadLocal<Map<String, String>> requestHeaders) {
+        this.requestHeaders = requestHeaders;
+    }
+
     public <T> T execute(Call<T> call) {
 
         final ResultCallback<T> resultCallback = new ResultCallback<T>();
 
+        requestHeaders.remove();
```

Comment:
Is this guaranteed to get called even if the request fails?

Reply:
Actually, the request has not been executed at this point. I think that's even better.